test/simulator/timestamp/src/timestamp_manager.cpp

```diff
@@ -188,3 +188,37 @@ timestamp_manager::validate_durable_ts(
 
     return (0);
 }
+
+/*
+ * Validate the read timestamp. The constraints on the read timestamp are:
+ * 1) The read timestamp can only be set before a transaction is prepared
+ * 2) Read timestamps can only be set once.
+ * 3) The read timestamp must be greater than or equal to the oldest timestamp unless rounding
+ * the read timestamp is enabled.
+ */
+int
+timestamp_manager::validate_read_timestamp(session_simulator *session, const uint64_t read_ts) const
+{
+    /* The read timestamp can't be set after a transaction is prepared. */
+    if (session->get_prepare_timestamp() != 0) {
```

Comment:
do we know whether a transaction is in prepared state or not with out checking prepare_ts ? my preference is to have __txn_prepared attribute similar to __txn_running attribute, if we already have that, than to use it here.

Reply:
I wasn't sure whether to include that in this ticket or in WT-9232 "Support prepare timestamp processing" or the other ticket for prepared transactions.